x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/User.java

```diff
@@ -192,41 +143,6 @@ public final XContentBuilder toXContent(XContentBuilder builder, Params params)
         return builder.endObject();
     }
 
-    public static User partialReadFrom(String username, StreamInput input) throws IOException {
-        String[] roles = input.readStringArray();
-        Map<String, Object> metadata = input.readMap();
-        String fullName = input.readOptionalString();
-        String email = input.readOptionalString();
-        boolean enabled = input.readBoolean();
-        User outerUser = new User(username, roles, fullName, email, metadata, enabled, null);
-        boolean hasInnerUser = input.readBoolean();
-        if (hasInnerUser) {
-            User innerUser = readFrom(input);
-            return new User(outerUser, innerUser);
-        } else {
-            return outerUser;
-        }
-    }
-
-    public static User readFrom(StreamInput input) throws IOException {
-        final boolean isInternalUser = input.readBoolean();
-        assert isInternalUser == false : "should always return false. Internal users should use the InternalUserSerializationHelper";
-        final String username = input.readString();
-        return partialReadFrom(username, input);
-    }
```

Comment:
This method is replaced by the new `AuthenticationSerializationHelper#readUserFrom`. The replacement is basically the old `InternalUserSerializationHelper.readFrom`.

Note that this method assert that the user cannot be internal. I don't think it is necessary at this method level. I have replaced it with assertions at the caller sites (there are only 2 of them).